### PR TITLE
Hyperscale backend is now supported. (asds distributed over all the storage nodes)

### PR DESCRIPTION
**Pull request by jeroenmaelbrancke**
Can someone review my code and give me some feedback?

### CODE REVIEW

File: ci/tests/general/general_alba.py [hunk 1]
```diff
@@ -334,12 +334,14 @@ def filter_disks(disk_names, amount, disk_type):
         :param disk_type: Type of disk
         :return: Filtered disks
         """
-        grid_ip = General.get_config().get('main', 'grid_ip')
-        storagerouter = GeneralStorageRouter.get_storage_router_by_ip(ip=grid_ip)
-        root_client = SSHClient(storagerouter, username='root')
-        hdds, ssds = GeneralDisk.get_physical_disks(client=root_client)
-        count = 0
-        filtered_disks = list()
+        hdds = dict()
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: We tend to use {} instead of dict()

File: ci/tests/general/general_alba.py [hunk 2]
```diff
@@ -334,12 +334,14 @@ def filter_disks(disk_names, amount, disk_type):
...
+        ssds = dict()
+        sorted_disks = dict()
+        filtered_disks = dict()
+
+        for node in AlbaNodeList.get_albanodes():
+            root_client = SSHClient(node.ip, username='root')
+            hdds[node.guid], ssds[node.guid] = GeneralDisk.get_physical_disks(client=root_client)
 
         if disk_type == 'SATA':
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: If i misspell SATA or SSD or use lowercase words i suddenly get results for all types

File: ci/tests/general/general_alba.py [hunk 3]
```diff
@@ -350,12 +352,27 @@ def filter_disks(disk_names, amount, disk_type):
             list_to_check = hdds.values()
 
         for disk_name in disk_names:
-            for disk in list_to_check:
-                if disk_name == disk['name']:
-                    filtered_disks.append(disk['name'])
-                    count += 1
-            if count == amount:
-                break
+            for node in list_to_check:
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: We seem to be using only the values for the physical disks, which was already a dict to begin with, can't this be optimized?

File: ci/tests/general/general_alba.py [hunk 4]
```diff
@@ -350,12 +352,27 @@ def filter_disks(disk_names, amount, disk_type):
...
+                for disk in node.values():
+                    if disk_name == disk['name']:
+                        if disk['ip'] in sorted_disks:
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: if disk['ip'] not in sorted_disks:
    sorted_disks[disk['ip']] = []
sorted_disks[disk['ip']].append...

File: ci/tests/general/general_alba.py [hunk 5]
```diff
@@ -350,12 +352,27 @@ def filter_disks(disk_names, amount, disk_type):
...
+                            sorted_disks[disk['ip']].append(disk['name'])
+                        else:
+                            sorted_disks[disk['ip']] = [disk['name']]
+
+        while amount > 0 and len(sorted_disks) > 0:
+            node = sorted_disks.keys()[amount % len(sorted_disks)]
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: Very unreadable and hard to understand code, can be optimized

File: ci/tests/general/general_alba.py [hunk 6]
```diff
@@ -350,12 +352,27 @@ def filter_disks(disk_names, amount, disk_type):
...
+            if len(sorted_disks[node]) == 0:
+                sorted_disks.pop(node)
+                continue
+            if node in filtered_disks:
+                filtered_disks[node].append(sorted_disks[node].pop())
+                amount -= 1
+                continue
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: obsolete continue

File: ci/tests/general/general_alba.py [hunk 7]
```diff
@@ -350,12 +352,27 @@ def filter_disks(disk_names, amount, disk_type):
...
+            else:
+                filtered_disks[node] = [sorted_disks[node].pop()]
+                amount -= 1
+                continue
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: obsolete continue

File: ci/tests/general/general_alba.py [hunk 8]
```diff
@@ -350,12 +352,27 @@ def filter_disks(disk_names, amount, disk_type):
...
-                break
+            for node in list_to_check:
+                for disk in node.values():
+                    if disk_name == disk['name']:
+                        if disk['ip'] in sorted_disks:
+                            sorted_disks[disk['ip']].append(disk['name'])
+                        else:
+                            sorted_disks[disk['ip']] = [disk['name']]
+
+        while amount > 0 and len(sorted_disks) > 0:
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: Entire loop can be optimized (i think)

File: ci/tests/general/general_alba.py [hunk 9]
```diff
@@ -381,11 +398,14 @@ def initialise_disks(alba_backend, nr_of_disks, disk_type):
         assert len(uninitialized_disk_names) >= nr_of_disks_to_init, "Not enough disks to initialize!"
 
         disks_to_init = GeneralAlba.filter_disks(uninitialized_disk_names, nr_of_disks_to_init, disk_type)
-        assert len(disks_to_init) >= nr_of_disks_to_init, "Not enough disks to initialize!"
-
-        grid_ip = General.get_config().get('main', 'grid_ip')
-        alba_node = AlbaNodeList.get_albanode_by_ip(grid_ip)
-        failures = AlbaNodeController.initialize_disks(alba_node.guid, disks_to_init)
+        amount_disks_to_init = 0
+        for ip in disks_to_init:
```
Comment (kvanhijf) [inline ci/tests/general/general_alba.py]: I'd expect to be able to initialize 1 disk on node1, 2 disks on node2, 3 disks on node3
Meaning, much more configurability

File: ci/tests/general/general_disk.py [hunk 10]
```diff
@@ -114,11 +114,13 @@ def get_physical_disks(client):
             disk_id = disk[0]
             if len(disk_id) > 2 and disk_id[0:2] in ['fd', 'sr', 'lo']:
                 continue
-            if disk[1] in 'disk':
+            types = ['disk', 'raid6']
+            disk[0] = disk[0].decode("ascii", errors="ignore")
```
Comment (kvanhijf) [inline ci/tests/general/general_disk.py]: Some comment line containing an example would be nice here why index 3 should '0', why index 1 needs to be 'disk' or 'raid6'

**Comment by pploegaert** [comment]
CLOSED: functionality moved to reworked autotests using scenarios

**Pull request closed without merging**